### PR TITLE
fix(environments): Handle the empty environment case in the stream

### PR DESCRIPTION
**Pull request by lynnagara**
Fix some bugs in how the stream view is handling a null and empty environment

### CODE REVIEW

File: src/sentry/static/sentry/app/views/stream/utils.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ function getQueryEnvironment(qs) {
 }
 
 function getQueryStringWithEnvironment(qs, env) {
-  const qsWithoutEnv = qs.replace(/environment:\w+/g, '');
+  const qsWithoutEnv = qs.replace(/environment:\w*/g, '');
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/stream/utils.js]: Handle when  `environment = ''"`  😭

File: src/sentry/static/sentry/app/views/stream.jsx [hunk 2]
```diff
@@ -538,7 +538,7 @@ const Stream = createReactClass({
 
       let query = streamUtils.getQueryStringWithEnvironment(
         this.state.query,
-        environment.name
+        environment === null ? null : environment.name
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/stream.jsx]: Forgot about `environment = null`

**Review by billyvg — APPROVED** [review]
Let's add test cases for these

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Pull request merged**